### PR TITLE
fix issue#2413: treat namespaces with exclude label as excludedNamespaces

### LINKED ISSUE #2413: Namespace with label velero.io/exclude-from-backup=true should exclude all resources within from backing up
**Describe the problem/challenge you have**
I want to have an option to exclude a namespace (all resources) without needing to edit velero config every time I create a new namespace. The actual option of `--exclude-namespaces` isn't enough because I don't want to create a naming pattern because of velero and since it already accept the label `velero.io/exclude-from-backup=true` to exclude resources dynamically why not treat everything from a namespace with that label as if all have that label? It just doesn't make sense for me to only skip the resource namespace backup but backup all the resources within it.

**Describe the solution you'd like**
A namespace with label `velero.io/exclude-from-backup=true` should be enough to exclude all resources in that namespace from velero backup without the need to add that label to all of them.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`):
```
Client:
	Version: v1.3.1
	Git commit: -
Server:
	Version: v1.3.1
```

- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

- Kubernetes installer & version:
```
kubeadm version: &version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:14:19Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):
```
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

### PR DESCRIPTION
**Pull request by allenxu404**
Thank you for contributing to Velero!

# Please add a summary of your change

- add namespaces with label velero.io/exclude-from-backup=true into request.Spec.ExcludedNamespaces, in this way, adding the label velero.io/exclude-from-backup=true to a namespace would be equivalent to setting spec.ExcludedNamespaces

# Does your change fix a particular issue?

Fixes #2413

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by blackpiglet — APPROVED** [review]

File: pkg/controller/backup_controller.go [hunk 1]
```diff
@@ -436,6 +436,15 @@ func (c *backupController) prepareBackupRequest(backup *velerov1api.Backup) *pkg
 	request.Annotations[velerov1api.SourceClusterK8sMajorVersionAnnotation] = c.discoveryHelper.ServerVersion().Major
 	request.Annotations[velerov1api.SourceClusterK8sMinorVersionAnnotation] = c.discoveryHelper.ServerVersion().Minor
 
+	// Add namespaces with label velero.io/exclude-from-backup=true into request.Spec.ExcludedNamespaces
+	// Essentially, adding the label velero.io/exclude-from-backup=true to a namespace would be equivalent to setting spec.ExcludedNamespaces
+	namespaces, excludeLabel := corev1api.NamespaceList{}, "velero.io/exclude-from-backup"
```
Comment (reasonerjt) [inline pkg/controller/backup_controller.go:441]: This is trivial but it seems we don't need `excludeLabel` as a variable.

File: pkg/controller/backup_controller.go [hunk 2]
```diff
@@ -436,6 +436,15 @@ func (c *backupController) prepareBackupRequest(backup *velerov1api.Backup) *pkg
...
+	if err := c.kbClient.List(context.Background(), &namespaces, kbclient.MatchingLabels{excludeLabel: "true"}); err == nil {
```
Comment (reasonerjt) [inline pkg/controller/backup_controller.go:442]: This is not a validation error strictly speaking but I don't think it's right to eat the error.
Would there be a problem if we set it into `request.Status.ValidationErrors` and make it fail early and explicitly?

File: pkg/controller/backup_controller.go (hunk 1 above)
Comment (allenxu404) [reply pkg/controller/backup_controller.go:441]: Would it be better if I move excludeLabel to pkg/controller/constants.go as a const or simply define it as const on the top of the code?

File: pkg/controller/backup_controller.go (hunk 2 above)
Comment (allenxu404) [reply pkg/controller/backup_controller.go:442]: Seems it's better to  append the potential err into request.Status.ValidationErrors with explicit err message.

**Review by reasonerjt — APPROVED** [review]
LGTM

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**